sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -906,20 +910,30 @@ static int sof_card_dai_links_create(struct device *dev,
 
 		ssp_components->name = codec_name;
 		ssp_components->dai_name = info->dai_name;
-		init_dai_link(links, id++, name, info->direction[0],
-			      info->direction[1], cpus, cpu_name,
+		cpus[cpus_id].dai_name = cpu_name;
+		init_dai_link(links, id, name, info->direction[0],
+			      info->direction[1], cpus + cpus_id, 1,
 			      ssp_components, 1, dai_id++, NULL, info->ops);
 
-		info->init(NULL, links + id - 1, info, 0);
+		info->init(NULL, links + id, info, 0);
+		cpus_id++;
+		id++;
 	}
 
 DMIC:
 	/* dmic */
 	if (dmic_num > 0) {
-		init_dai_link(links, id++, "dmic01", 0, 1, cpus, "DMIC01 Pin",
+		cpus[cpus_id].dai_name = "DMIC01 Pin";
+		init_dai_link(links, id, "dmic01", 0, 1, cpus + cpus_id, 1,
```

Comment:
Btw, why are we hard coding DAI names ? Shouldn't  these all be coming from topology ?

Reply:
@lgirdwood The topology and hard-coded BE dai-link info need to be aligned.
Nothing new here, this is how we've done things from the beginning

Reply:
ok, but how do we create "DMIC02 Pin" or "dmic02"

Reply:
we only have DMIC01 and DMIC16k

https://github.com/thesofproject/linux/blob/8d47294d6a699a1139f080ae9ac0ddfc790cc527/sound/soc/sof/intel/hda-dai.c#L442